### PR TITLE
Neo request stats

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Oh, needs release notes.

**Comment by nemanja-boric-sociomantic** [comment]
Travis says no.

**Comment by gavin-norman-sociomantic** [comment]
Updated:
* Fixed D2 compilation error.
* Added tests for neo request stats logging.
* Added release notes.

**Comment by gavin-norman-sociomantic** [comment]
Rebased.

**Comment by gavin-norman-sociomantic** [comment]
Rebased.

File: src/swarm/node/model/IChannelsNodeInfo.d [hunk 1]
```diff
@@ -174,13 +174,15 @@ version ( UnitTest )
...
                 actions = list of record action types to track stats for
                 requests = list of request types to track stats for
+                neo_requests = list of neo request types to track stats for
                 channels = list of channels to track stats for
 
         ***********************************************************************/
 
-        this ( istring[] actions, istring[] requests, istring[] channels )
+        this ( istring[] actions, istring[] requests, istring[] neo_requests,
+            istring[] channels )
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/IChannelsNodeInfo.d:183]: Is this a breaking change?

Comment (gavin-norman-sociomantic) [reply src/swarm/node/model/IChannelsNodeInfo.d:183]: I think so. Well spotted. I'll have a look at it.

Comment (gavin-norman-sociomantic) [reply src/swarm/node/model/IChannelsNodeInfo.d:183]: Ah no. It's just a class used internally within swarm for unittesting. Of course, strictly speaking, code outside of swarm _could_ use this, but that would be very weird and is not intended.

**Review by david-eckardt-sociomantic — APPROVED** [review]

**Pull request merged**